pymc/gp/cov.py

```diff
@@ -71,29 +78,29 @@ def __call__(self, X, Xs=None, diag=False):
         else:
             return self.full(X, Xs)
 
-    def diag(self, X):
+    def diag(self, X: TensorLike) -> TensorVariable:
         raise NotImplementedError
 
-    def full(self, X, Xs=None):
+    def full(self, X: TensorLike, Xs: Optional[TensorLike] = None) -> TensorVariable:
         raise NotImplementedError
 
-    def __add__(self, other):
+    def __add__(self, other) -> "Add":
```

Comment:
might be an ignorant question, but why is the return type a string instead of `Add`?

Reply:
Basically because the `Add` class has not yet been defined (as it is actually a subclass of this one), therefore, as per [PEP 484](https://peps.python.org/pep-0484/#forward-references) you can define it as a string instead and mypy will work accordingly. Though I *think* as of 3.10 you should be able to specify without the string.